File: monolith-spigot/src/main/java/gg/warcraft/monolith/spigot/world/SpigotSoundMapper.java
package gg.warcraft.monolith.spigot.world;

import gg.warcraft.monolith.api.world.Sound;
import gg.warcraft.monolith.api.world.SoundCategory;

public class SpigotSoundMapper {

    public org.bukkit.Sound map(Sound sound) {
        return org.bukkit.Sound.valueOf(sound.name());
    }

    public Sound map(org.bukkit.Sound sound) {
        return Sound.valueOf(sound.name());
    }

    public org.bukkit.SoundCategory map(SoundCategory category) {
        return org.bukkit.SoundCategory.valueOf(category.name());
    }

    public SoundCategory map(org.bukkit.SoundCategory category) {
        return SoundCategory.valueOf(category.name());
    }
}

File: monolith-spigot/src/main/scala/gg/warcraft/monolith/spigot/world/SpigotSoundMapper.scala
package gg.warcraft.monolith.spigot.world

import gg.warcraft.monolith.api.world.{Sound, SoundCategory}
import org.bukkit.{Sound => SpigotSound, SoundCategory => SpigotSoundCategory}

class SpigotSoundMapper {
  def map(sound: SpigotSound): Sound =
    Sound.valueOf(sound.name())

  def map(sound: Sound): SpigotSound =
    SpigotSound.valueOf(sound.name())

  def map(category: SpigotSoundCategory): SoundCategory =
    SoundCategory.valueOf(category.name())

  def map(category: SoundCategory): SpigotSoundCategory =
    SpigotSoundCategory.valueOf(category.name())
}
